### PR TITLE
[feature] Added user info to status page #104

### LINKED ISSUE #104: [wifi-login-pages] Feature: show user details in status page
In the status page, we need to show the details of the currently logged in user:

- username (only if different than email and phone number)
- email
- phone number (if mobile phone verification enabled)

### PR DESCRIPTION
**Pull request by NoumbissiValere**
Closes #104

### CODE REVIEW

**Comment by NoumbissiValere** [comment]
@nemesisdesign the complain is not part of this solution but I don't get it why linting is complaining in [phone-token-controller](https://github.com/openwisp/openwisp-wifi-login-pages/blob/master/server/controllers/mobile-phone-token-controller.js#L23) and not in [phone change number](https://github.com/openwisp/openwisp-wifi-login-pages/blob/master/server/controllers/mobile-phone-number-change-controller.js#L24) for the same thing.

**Comment by NoumbissiValere** [comment]
[PR 202](https://github.com/openwisp/openwisp-radius/pull/202) is needed to test this work

File: client/components/status/status.js [hunk 1]
```diff
@@ -188,8 +199,17 @@ export default class Status extends React.Component {
           '"response_code" !== "AUTH_TOKEN_VALIDATION_SUCCESSFUL"',
         );
       } else {
-        const {radius_user_token: password, username, is_active} = response.data;
-        this.setState({username, password, is_active});
+        const {radius_user_token: password, username, email, phone_number, is_active} = response.data;
+        const user_info = {};
```
Comment (nemesisdesign) [inline client/components/status/status.js:203]: use camel case in javascript code please

File: client/components/status/status.js [hunk 2]
```diff
@@ -211,23 +231,25 @@ export default class Status extends React.Component {
...
-              })}
+              <table>
+                <tbody>
+                  {Object.entries(user_info).map(([key, value]) => {
+                      return (
+                        <tr key={key}>
+                          <td>{key}:</td>
+                          <td>{value}</td>
+                        </tr>
+                      );
```
Comment (nemesisdesign) [inline client/components/status/status.js:249]: Please define the fields one by one and get the translatable text in a way that is consistent with the rest of the application (create one config key for each word).

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Good progress @NoumbissiValere, here's a quick review, will need to look more in depth into it tomorrow and test it as well.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
- The status row indicated in the wireframe is missing.
- I would switch from using a table to using a series of `<p>` elements, `<label>` and span, eg:
`<p><label>Email:</label><span>email value</span></p>`
- I also suggest using the CSS of the component to style the label to be bold and to have some margin on the right side.
- Please when you need to add space between elements use CSS, not <br> elements, the space below is huge and looks bad:

Screenshot from 2020-12-09 21-41-20

**Comment by NoumbissiValere** [comment]
@nemesisdesign should I remove this text:
```WiFi Login Successful!

You can now use the internet.

You may leave this page open in case you want to log out.
```

**Comment by nemesisdesign** [comment]
@NoumbissiValere leave it for now, I will think about it a bit more.

File: .eslintrc.json [hunk 3]
```diff
@@ -21,6 +21,7 @@
     "node": true
   },
   "rules": {
+    "jsx-a11y/label-has-associated-control": 0,
```
Comment (nemesisdesign) [inline .eslintrc.json:24]: adding this rule here affects the whole project. It should be possible to add an ignore rule only in the interested file, which is a lot better.

File: client/components/status/status.js [hunk 4]
```diff
@@ -211,23 +221,32 @@ export default class Status extends React.Component {
...
-                return null;
-              })}
+                  if (text !== "")
+                    return (
+                      <p key={text}>
+                        {text}
+                      </p>
+                    );
+                  return null;
+                })}
```
Comment (nemesisdesign) [inline client/components/status/status.js:241]: why did you increase the indentation here? It looks like an unnecessary change and I think it should be reverted.

File: internals/generators/organization/config.yml.hbs [hunk 5]
```diff
@@ -281,6 +281,23 @@ client:
...
+          text:
+            en: "Email"
+        username:
+          text:
+            en: "Username"
+      {{#if mobile_phone_verification}}
+        phone_number:
+          text:
+            en: "Phone number"
+      {{/if}}
```
Comment (nemesisdesign) [inline internals/generators/organization/config.yml.hbs:300]: let's not use the if here, the reason is that if mobile phone verification is enabled at a later step, the user will have to modify manually the config file which is time consuming and error prone.

File: org-configurations/default-configuration.yml [hunk 6]
```diff
@@ -208,6 +208,18 @@ client:
...
+          text:
+            en: "Status"
+          value:
+            en: "Logged in"
+        email:
+          text:
+            en: "Email"
+        username:
+          text:
+            en: "Username"
```
Comment (nemesisdesign) [inline org-configurations/default-configuration.yml:222]: please add the mobile phone field here as well for consistency with the change requested above

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Tested and looks good, please address the following concerns below so we can merge.

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**